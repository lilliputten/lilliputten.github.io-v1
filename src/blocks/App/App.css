// vim: ft=scss

.App {
  // display: flex;
  // flex-direction: row;
  // align-items: center;
  &-Header {
    text-align: center;
  }
  &-Title {
    font-size: 1.25em;
  }
  &-Intro {
    font-size: large;
  }
}
