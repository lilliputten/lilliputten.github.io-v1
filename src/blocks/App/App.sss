
.App
  text-align: center;
  border: 1px solid red
  display: flex
  flex-direction: row
  align-items: center

// .App-Title {
//   font-size: 1.5em;
// }
//
// .App-Intro {
//   font-size: large;
// }
