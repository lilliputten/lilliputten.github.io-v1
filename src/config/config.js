/**
 * @module config
 * @overview Application config
 * @author lilliputten <lilliputten@yandex.ru>
 * @since 2018.03.10, 02:05
 * @version 2018.03.10, 23:02
 */

import site from './site'

const testMode = (typeof jest === 'object');
const hotMode = !!(typeof module === 'object' && module.hot && module.hot.active);
const DEBUG = testMode ? 'test' : hotMode;

const config = /** @lends config */{

  /** Is test mode? */
  testMode : testMode,
  /** Is hot dev-server mode? */
  hotMode : hotMode,
  /** Is DEBUG mode? (webpack server in hot mode) */
  DEBUG : DEBUG,

  // App start time (DEBUG?)
  startTime : Date.now(), // ( DEBUG === 'test' ) ? 0 : Date.now(),

  /** Site properties */
  site : site,

};

export default config;

