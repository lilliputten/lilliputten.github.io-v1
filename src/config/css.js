/**
 * @module css
 * @overview Export configuration for system styles
 * @author lilliputten <lilliputten@yandex.ru>
 * @since 2018.03.22, 23:39
 * @version 2018.03.22, 23:41
 */

const primaryColor = '#c10';
const defaultFontSize = 16;

/** cssVariables ** {{{ */
const cssVariables = {

  primaryColor,

  // DEBUG...

  testGlobalColor: 'orange', // test vatiable

  testViewportWidth: 300, // test media size

  // Fonts...

  defaultFontSize,
  fontSize: defaultFontSize,
  fontSizeSm: defaultFontSize - 2,
  fontSizeXs: defaultFontSize - 4,
  fontSizeLg: defaultFontSize + 2,
  fontSizeXl: defaultFontSize + 4,
  fontSizeXxl: defaultFontSize + 6,

  defaultFontFamily: '-apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, "Helvetica Neue", Arial, sans-serif',
  defaultFontWeight: 400,
  defaultLineHeight: 1.5, // 1.5 ??? too loose? (from mdbootstrap)

  defaultTitleFontSize: 28,
  defaultTitleFontWeight: 300,

  // Page...

  pageBgColor: '#fff',

  defaultTextColor: '#333',
  defaultLinkColor: primaryColor,
  defaultHoveredLinkColor: '#f10',

  // Navbar...

  NavbarHeight: 50,

  NavBarTreshold: 600,

};/*}}}*/

// Using classic export due to requires in `config-override.js`
module.exports = cssVariables;
