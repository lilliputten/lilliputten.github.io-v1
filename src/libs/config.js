/**
 * @module config
 * @overview Application config
 * @author lilliputten <lilliputten@yandex.ru>
 * @since 2018.03.10, 02:05
 * @version 2018.03.10, 23:02
 */

const testMode = (typeof jest === 'object');
const hotMode = !!(typeof module === 'object' && module.hot && module.hot.active);
const DEBUG = testMode ? 'test' : hotMode;

const config = {

  // Test mode?
  testMode : testMode,
  // Hot dev-server mode?
  hotMode : hotMode,
  // Development mode? (webpack server in hot mode)
  DEBUG : DEBUG,

  // Site content root url prefix
  siteRootPrefix : '/site',

  // App start time
  startTime : Date.now(), // ( DEBUG === 'test' ) ? 0 : Date.now(),

};

export default config;

