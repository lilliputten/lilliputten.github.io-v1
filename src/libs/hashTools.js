/**
 * @module hashTools
 * @author lilliputten <lilliputten@yandex.ru>
 * @since 2018.03.11, 20:32
 * @version 2018.03.11, 20:32
 */

import config from 'config';

const hashTools = {

  /** getFromWindow ** {{{ Get hash from window location */
  getFromWindow : function () {
    return ( typeof window === 'object' && window.location && window.location.hash ) || '';
  },/*}}}*/

  /** setToWindow ** {{{ Set hash to window location
   * @param {String} hash
   */
  setToWindow : function (hash) {
    if ( typeof window === 'object' && window.location ) {
      window.location.hash = hash;
    }
  },/*}}}*/

  /** getPageIdFromWindow ** {{{ Get page id from window location hash
   * @return {String}
   */
  getPageIdFromWindow : function () {
    const hash = this.getFromWindow();
    if ( this.isCorrectHash(hash) ) {
      return this.getPageId(hash);
    }
    return '';
  },/*}}}*/

  /** isCorrecthash ** {{{ Check for correct hash value (empty or in form `#!string/hash`)
   * @param {String} hash
   * @return {Boolean}
   */
  isCorrectHash : function (hash) {

    return !!( hash != null && ( hash === '' || hash.startsWith('#!') || hash.startsWith('#') ) );

  },/*}}}*/

  /** toPageId ** {{{ Convert hash to page id (w/o root & extension)
   * @param {String} hash
   * @return {String} pageId
   */
  toPageId : function (hash) {

    // Strip leading '#!'
    if (hash.startsWith('#')) {
      hash = hash.substr(1);
    }
    // ??
    if (hash.startsWith('!')) {
      hash = hash.substr(1);
    }
    // ???
    if (hash.startsWith('/')) {
      hash = hash.substr(1);
    }

    return hash;

  },/*}}}*/

  /** toUrl ** {{{ Convert hash to base url (w/o extension)
   * @param {String} hash
   * @param {String} [extension]
   * @return {String} url
   */
  toUrl : function (hash, extension) {

    let page = this.toPageId(hash);

    // Adding leading slash
    if (!page.startsWith('/')) {
      page = '/' + page;
    }

    // Adding index if ends with '/'
    if (page.endsWith('/')) {
      page += config.site.defaultIndex;
    }

    // Adding extension
    if (extension) {
      page += extension;
    }

    return config.site.rootPrefix + page;

  },/*}}}*/

  /** fromUrl ** {{{ Convert url to hash
   * @param {String} url
   * @return {String} hash
   */
  fromUrl : function (url) {

    let hash = url;

    // Hash at root url
    if (hash.startsWith('/#!')) {
      return hash.substr(1);
    }

    // Removing default extensions and index
    const stripAtEnd = config.site.defaultExts.concat(config.site.defaultIndex);
    stripAtEnd.map((tail) => {
      if (hash.endsWith(tail)) {
        hash = hash.substr(0, hash.length - tail.length);
      }
      return hash;
    });

    // Removing root prefix
    let found = false;
    if (hash.startsWith(config.site.rootPrefix)) {
      hash = hash.substr(config.site.rootPrefix.length);
      found = true;
    }
    // Removing leading slash
    if (hash.startsWith('/')) {
      hash = hash.substr(1);
      found = true;
    }
    // Invalid url
    if (!found) {
      return null;
    }

    if (this.isCorrectHash(hash)){
      hash = this.toPageId(hash);
    }

    // Return empty value if default page
    if (hash === config.site.defaultPage) {
      hash = '';
    }

    return hash && '#!' + hash;

  },/*}}}*/

  /** getPageId ** {{{ Get page id from hash/url
   * @param {String} urlOrHash - Hash (#!/dir/page) or url (<rootPrefix>/dire/page)
   * NOTE: Highest level method on `toPageId`.
   * @return {null|String) - Return null if can't fing page id.
   */
  getPageId : function (urlOrHash) {

    if (urlOrHash == null) {
      return null;
    }
    else if (urlOrHash === '') {
      return config.site.defaultPage;
    }
    else {
      if (urlOrHash.startsWith(config.site.rootPrefix)) {
        urlOrHash = this.fromUrl(urlOrHash);
      }
      if (this.isCorrectHash(urlOrHash)) {
        const pageId = this.toPageId(urlOrHash);
        if (!pageId || pageId === config.site.defaultPage) {
          return '';
        }
        return pageId;
      }
    }

    return null;

  },/*}}}*/

  /** fromPageId ** {{{ Get hash for pageId
   * @param {String} pageId
   * @return {String} hash
   */
  fromPageId : function (pageId) {
    return '#!' + pageId;
  },/*}}}*/

  /** isDefaultHash ** {{{ Is hash for default page?
   * @param {String} hash - Hash (#!/dir/page)
   * @return {Boolean}
   */
  isDefaultHash : function (hash) {
    return hash === this.fromPageId(config.site.defaultPage);
  },/*}}}*/

};

export default hashTools;


