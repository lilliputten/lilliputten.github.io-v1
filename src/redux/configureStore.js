/**
 * @module configureStore
 * @author lilliputten <lilliputten@yandex.ru>
 * @since 2018.03.18, 02:42
 * @version 2018.03.19, 23:52
 */

import config from 'config'
import { applyMiddleware, combineReducers, compose, createStore } from 'redux'
import thunk from 'redux-thunk'
import pageReducer from './reducers/pageReducer'

const rootReducer = combineReducers({
  page : pageReducer
});

const noop = () => {};

const enhancer = compose(
  applyMiddleware(thunk),
  // Adding Redux DevTools
  (config.DEBUG && window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()) || noop
);

export default function configureStore(initialState = {}) {
  return createStore(rootReducer, initialState, enhancer);
}
