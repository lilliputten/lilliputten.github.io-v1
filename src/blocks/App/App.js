/**
 * @module App
 * @author lilliputten <lilliputten@yandex.ru>
 * @since 2018.01.28, 23:51
 * @version 2018.03.25, 07:13
 */

import React, { Fragment } from 'react'
import { decl/* , Bem */ } from 'bem-react-core'
import { connect } from 'react-redux'
import hashTools from 'libs/hashTools'

import config from 'config'

import Header from 'e:Header'
import View from 'b:View'

const App_proto = /** @lends App.prototype */{

  block: 'App',

  /** willInit ** {{{ */
  willInit() {

    const hash = hashTools.getFromWindow();
    const page = hashTools.getPageId(hash) || config.site.defaultPage;

    this.state = {
      page,
    };

    this.props.store && this.props.store.subscribe(() => {
      const state = this.props.store.getState();
      const page = state.page.page;
      this.setState({
        page,
      });
    });

  },/*}}}*/

  /** mods ** {{{ Modifiers... */
  mods(self) {
    const page = this.state.page;
    return { ...self.mods,
      page,
    };
  },/*}}}*/

  /** content ** {{{ */
  content() {
    return (
      <Fragment>
        <Header/>
        <View loadPage hashChange clickHandle store={this.props.store} />
      </Fragment>
    );
  },/*}}}*/

};

const mapStateToProps = (state) => {
  const { page } = state.page || {};
  return { page };
}

export default decl(App_proto, connect(mapStateToProps));

