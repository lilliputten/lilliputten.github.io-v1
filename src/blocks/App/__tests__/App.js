import React from 'react'

import axios from 'axios'
import MockAdapter from 'axios-mock-adapter'

import Enzyme, { mount/* , shallow */ } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import config from 'config'
import hashTools from 'libs/hashTools'

import App from 'b:App'

import configureStore from 'redux/configureStore'

const mock = new MockAdapter(axios);

const initialState = window.REDUX_INITIAL_STATE || {};
const store = configureStore(initialState);

Enzyme.configure({ adapter: new Adapter() })

describe('App', () => {

  let appWrapper, app;

  // Default page...
  const pageId = config.site.defaultPage;
  const pageHash = hashTools.fromPageId(pageId);
  const pageUrl = hashTools.toUrl(pageHash, config.site.defaultExt);
  const paramsUrl = hashTools.toUrl(pageHash, config.site.dataExt);
  const pageBody = 'page content';
  // const pageHtml = '<p>' + pageBody + '</p>\n';
  const paramsBody = '{"test":"ok"}';
  // const paramsData = { test : 'ok' };

  // Default page mocks...
  mock.onAny(pageUrl).reply(200, pageBody);
  mock.onAny(paramsUrl).reply(200, paramsBody);

  beforeAll(() => {
    appWrapper = mount(
      <App store={store} />
    );
    app = appWrapper.instance();
  })

  afterAll(() => {
    appWrapper.unmount();
  })

  it('match snapshot', () => {
    expect(appWrapper).toMatchSnapshot();
  })

  it('contains View', () => {
    // get ReactWrapper object by querySelector
    const childWrapper = appWrapper.find('.View');
    expect(childWrapper).toHaveLength(1);
  })

  it('is instance of React.Component', () => {
    expect(app).toBeInstanceOf(React.Component);
  })

})
