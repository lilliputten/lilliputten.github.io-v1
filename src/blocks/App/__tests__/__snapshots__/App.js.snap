// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App match snapshot 1`] = `
<Connect(App)
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <App
    dispatch={[Function]}
    page="default"
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
    storeSubscription={
      Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": [Function],
      }
    }
  >
    <div
      className="App App_mode_default"
      style={Object {}}
    >
      <App-Spinner>
        <div
          className="App-Spinner"
          style={Object {}}
        >
          <Bem
            alt="Loading..."
            elem="SpinnerImg"
            src="spinner-dots.gif"
            tag="img"
          >
            <img
              alt="Loading..."
              className="App-SpinnerImg"
              src="spinner-dots.gif"
            />
          </Bem>
        </div>
      </App-Spinner>
      <Connect(Navbar)
        mode="default"
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <Navbar
          dispatch={[Function]}
          mode="default"
          page="default"
          store={
            Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            }
          }
          storeSubscription={
            Subscription {
              "listeners": Object {
                "clear": [Function],
                "get": [Function],
                "notify": [Function],
                "subscribe": [Function],
              },
              "onStateChange": [Function],
              "parentSub": undefined,
              "store": Object {
                "dispatch": [Function],
                "getState": [Function],
                "replaceReducer": [Function],
                "subscribe": [Function],
                Symbol(observable): [Function],
              },
              "unsubscribe": [Function],
            }
          }
        >
          <div
            className="Navbar Navbar_mode_default"
            style={Object {}}
          >
            <Navbar-Logo>
              <div
                className="Navbar-Logo"
                style={Object {}}
              >
                <Bem
                  elem="LogoLink"
                  href="#!"
                  tag="a"
                >
                  <a
                    className="Navbar-LogoLink"
                    href="#!"
                  >
                    <Bem
                      alt="logo"
                      elem="LogoImg"
                      src="Logo.svg"
                      tag="img"
                    >
                      <img
                        alt="logo"
                        className="Navbar-LogoImg"
                        src="Logo.svg"
                      />
                    </Bem>
                  </a>
                </Bem>
              </div>
            </Navbar-Logo>
            <Navbar-Menu>
              <div
                className="Navbar-Menu"
                style={Object {}}
              >
                <Navbar-MenuItem
                  key="#!about"
                  text="About"
                  url="#!about"
                >
                  <a
                    className="Navbar-MenuItem"
                    href="#!about"
                    style={Object {}}
                  >
                    About
                  </a>
                </Navbar-MenuItem>
                <Navbar-MenuItem
                  key="#!projects"
                  text="Projects"
                  url="#!projects"
                >
                  <a
                    className="Navbar-MenuItem"
                    href="#!projects"
                    style={Object {}}
                  >
                    Projects
                  </a>
                </Navbar-MenuItem>
                <Navbar-MenuItem
                  key="#!contacts"
                  text="Contacts"
                  url="#!contacts"
                >
                  <a
                    className="Navbar-MenuItem"
                    href="#!contacts"
                    style={Object {}}
                  >
                    Contacts
                  </a>
                </Navbar-MenuItem>
              </div>
            </Navbar-Menu>
          </div>
        </Navbar>
      </Connect(Navbar)>
      <Connect(View)
        clickHandle={true}
        hashChange={true}
        loadPage={true}
        mode="default"
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <View
          clickHandle={true}
          dispatch={[Function]}
          hashChange={true}
          loadPage={true}
          mode="default"
          page="default"
          store={
            Object {
              "dispatch": [Function],
              "getState": [Function],
              "replaceReducer": [Function],
              "subscribe": [Function],
              Symbol(observable): [Function],
            }
          }
          storeSubscription={
            Subscription {
              "listeners": Object {
                "clear": [Function],
                "get": [Function],
                "notify": [Function],
                "subscribe": [Function],
              },
              "onStateChange": [Function],
              "parentSub": undefined,
              "store": Object {
                "dispatch": [Function],
                "getState": [Function],
                "replaceReducer": [Function],
                "subscribe": [Function],
                Symbol(observable): [Function],
              },
              "unsubscribe": [Function],
            }
          }
        >
          <div
            className="View View_mode_loading"
            style={Object {}}
          >
            <Bem
              elem="ContentWrapper"
              mode="loading"
            >
              <div
                className="View-ContentWrapper"
                mode="loading"
              >
                <Bem
                  elem="Status"
                  tag="p"
                >
                  <p
                    className="View-Status"
                  />
                </Bem>
              </div>
            </Bem>
          </div>
        </View>
      </Connect(View)>
      <Footer
        mode="default"
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <div
          className="Footer Footer_mode_default"
          style={Object {}}
        >
          ...
        </div>
      </Footer>
    </div>
  </App>
</Connect(App)>
`;
